.azure/pipelines/ci.yml

```diff
@@ -582,9 +581,16 @@ stages:
       agentOs: Windows
       timeoutInMinutes: 180
       steps:
-      - script: .\restore.cmd -ci
-        displayName: Restore
-      - script: .\build.cmd -ci -NoRestore -test -projects eng\helix\helix.proj /p:IsRequiredCheck=true /p:IsHelixJob=true /p:BuildAllProjects=true /p:BuildNative=true /p:ASPNETCORE_TEST_LOG_DIR=artifacts/log -bl
+      # Build the shared framework
+      - script: ./build.cmd -ci -all -pack -arch x64 /p:ASPNETCORE_TEST_LOG_DIR=artifacts/log /bl:artifacts/log/helix.build.x64.binlog
+        displayName: Build shared fx
+      # Build the x86 shared framework
+      # This is going to actually build x86 native assets.
+      - script: ./build.cmd -ci -arch x86 -pack -all -buildNative -noBuildJava /p:OnlyPackPlatformSpecificPackages=true /p:ASPNETCORE_TEST_LOG_DIR=artifacts/log /bl:artifacts/log/helix.build.x86.binlog $(_BuildArgs)
+        displayName: Build x86 shared framework pieces
+      - script: ./src/ProjectTemplates/build.cmd -ci -pack -NoRestore -NoBuilddeps "/p:ASPNETCORE_TEST_LOG_DIR=artifacts/log /bl:artifacts/log/helix.template.pack.binlog"
+        displayName: Pack Templates        
+      - script: .\build.cmd -ci -noBuildRepoTasks -noRestore -noBuild -test -projects eng\helix\helix.proj /p:IsRequiredCheck=true /p:IsHelixJob=true /p:BuildAllProjects=true /p:BuildNative=true /p:ASPNETCORE_TEST_LOG_DIR=artifacts/log -bl
```

Comment:
@HaoK can you try removing -noBuild to see if IIS tests work? I believe there is a msbuild target that runs on build which creates "published" outputs. https://github.com/dotnet/aspnetcore/blob/f676c249d25eb438ffc282edb551d86eea1d9709/src/Servers/IIS/IIS/test/FunctionalTest.props#L13-L29

Reply:
In the IIS tests results, there should be a folder called InProcessWebSite-Portable, that isn't present in your PR.

Reply:
https://helix.dot.net/api/2019-06-17/jobs/98723391-b094-4390-ab00-e6a95a3005b1/workitems/IIS.FunctionalTests--netcoreapp5.0/console

Reply:
Okay that makes sense, I'll give it a shot

Reply:
That was it, thanks!

Reply:
Does calling build cause the build times to increase? Maybe we can fix the target to copy after a different event?

Reply:
Not really an issue yet, mac and test templates are still at 1:24, so the helix step is still around 1:18

Reply:
But yeah we should probably consolidate the shared framework building in one central place rather than doing it in 3-4 different jobs.  I don't think that needs to be done as part of this PR though.